### PR TITLE
Wiimote fixes

### PR DESCRIPTION
**Pull request by bryangrim**
These changes allow for:
 * Querying Bluetooth devices using LIAC mode, which is the only way I could discover my third-party wiimotes.
 * Checking whether existing wiimotes are already in use, so that one wiimote does not fill all 4 spots in the global array (which was happening to me)

Note, there are still some odd things about how these wiimotes work, but I think it is related to the poor-quality wiimotes.  Things like:
 * sometimes needing to do a reset of the controller 5-10 minutes into a game after it worked perfectly prior to it shutting off.
 * sometimes the wiimote gets recognized by dolphin but the lights don't stop flashing till an emulation is started

### CODE REVIEW

File: Source/Core/Core/HW/WiimoteReal/IONix.cpp [hunk 1]
```diff
@@ -82,9 +84,17 @@ void WiimoteScanner::FindWiimotes(std::vector<Wiimote*> & found_wiimotes, Wiimot
 	inquiry_info scan_infos[max_infos] = {};
 	auto* scan_infos_ptr = scan_infos;
 	found_board = nullptr;
+	// query in limited IAC - liac - using code from hcitool
+	// utils/tools/hcitool.c
+	int l;
+	uint8_t lap[3] = { 0x33, 0x8b, 0x9e };
+	l = 0x9e8b00; // LIAC (https://www.bluetooth.org/en-us/specification/assigned-numbers/baseband)
```
Comment (lioncash) [inline Source/Core/Core/HW/WiimoteReal/IONix.cpp]: const int liac_value = 0x9e8b00;

File: Source/Core/Core/HW/WiimoteReal/IONix.cpp [hunk 2]
```diff
@@ -15,6 +15,7 @@
 
 namespace WiimoteReal
 {
+extern Wiimote* g_wiimotes[MAX_BBMOTES];
```
Comment (delroth) [inline Source/Core/Core/HW/WiimoteReal/IONix.cpp]: Please define that in the appropriate .h file and include it here instead. Cross-dependencies across compilation units like this are fragile.

Comment (delroth) [inline Source/Core/Core/HW/WiimoteReal/IONix.cpp]: (Preferably: redesign the API so the dependency on global state is not required.)

File: Source/Core/Core/HW/WiimoteReal/IOWin.cpp [hunk 3]
```diff
@@ -789,6 +790,10 @@ int WiimoteWindows::IOWrite(const u8* buf, size_t len)
 {
 	return _IOWrite(m_dev_handle, m_hid_overlap_write, m_stack, buf, len, nullptr);
 }
+char* WiimoteWindows::Address()
+{
+        return m_devicepath.c_str();
```
Comment (delroth) [inline Source/Core/Core/HW/WiimoteReal/IOWin.cpp]: That returns a const char*. You lose the const safety here, I'm surprised it even compiles.

File: Source/Core/Core/HW/WiimoteReal/WiimoteReal.h [hunk 4]
```diff
@@ -62,6 +62,7 @@ friend class WiimoteEmu::Wiimote;
 	// these are called from the Wiimote's thread.
 	virtual bool ConnectInternal() = 0;
 	virtual void DisconnectInternal() = 0;
+	virtual char* Address() = 0;
```
Comment (delroth) [inline Source/Core/Core/HW/WiimoteReal/WiimoteReal.h]: Why not use std::string?

**Comment by delroth** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/Core/HW/WiimoteReal/IONix.cpp (hunk 2 above)
Comment (bryangrim) [inline Source/Core/Core/HW/WiimoteReal/IONix.cpp]: Just going to ask if that was an OK alternative - thanks!

**Comment by bryangrim** [comment]
Not sure the proper process for fixing these comments.  I've reworked things and pushed it to this branch.  If there's anything else I'm supposed to do to get things moving, please let me know.

**Comment by lioncash** [comment]
Please squash your changes into one commit.

**Comment by bryangrim** [comment]
Hopefully I squashed things properly... (sorry, new to Github).

**Comment by lioncash** [comment]
unfortunately not

**Comment by degasus** [comment]
My real wiimote does still work with this patch.

**Comment by Phatcat** [comment]
@bryangrim

A somehow short and easy way to squash all (or a select number of the most recent) commits on a pr into one;
Open the related branch in Git Shell (if you are using the GitHub Desktop GUI client, you want to go to the branch, then press the options 'gear' icon in the upper-right corner, and choose 'Open in Git Shell'), and enter the following command 'git reset --soft HEAD~n', where n is the nuber of commits you want to reset, in your case 5:

- git reset --soft HEAD~5

Now your 5 latest commits to this branch have been revoked, with the file changes still present on your local copy however. What you wanna do now is make a new commit containing all the file changes, so you type the following command:

- git commit

(Or, if you are on the GitHub client, you can navigate back to the main program window and see that all the files changed are now present as if you never commited any of them. Now select which files (likely all) you want to commit and write a title and description and then press 'Commit to ....' [where .... is the name of the branch] - Do NOT press 'Sync' as you normally would, though!)

This now makes a new, single commit, based on all the work in your 5 commits combinded. This has however, only changed things locally, so now you have to force push the changes to your remote branch. If you don't force push the changes, chances are that it will instead register no change (the 5 old commits on remote and your 1 new commit on local is identical in files edited), and then pull back the old commits to your local copy, instead of pushing the new commit, so you want to enter the following command:

- git push -f

This will force the local branch to push the changes to the remote branch, which effectively removes the 5 old commits from the remote branch and replaces them with the 1 new commit.

Hope this helped.

**Comment by bryangrim** [comment]
Thanks @Phatcat.  Unfortunately, I had already messed things up enough on my side that things weren't working.  I hope it's not a pain that I created a new branch/pull request ( #3327 ).  I'll close this one shortly.

**Pull request closed without merging**